packages/auth/plugin/src/ios/openUrlFix.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  ConfigPlugin,
+  withAppDelegate,
+  withInfoPlist,
+  ExportedConfigWithProps,
+  WarningAggregator,
+} from '@expo/config-plugins';
+import type { ExpoConfig } from '@expo/config/build/Config.types';
+import type { AppDelegateProjectFile } from '@expo/config-plugins/build/ios/Paths';
+import type { InfoPlist } from '@expo/config-plugins/build/ios/IosConfig.types';
+import { mergeContents } from '@expo/config-plugins/build/utils/generateCode';
+import { PluginConfigType } from '../pluginConfig';
+
+export const withIosCaptchaOpenUrlFix: ConfigPlugin<PluginConfigType> = (
+  config: ExpoConfig,
+  props?: PluginConfigType,
+) => {
+  // check configuration
+  if (!shouldApplyIosOpenUrlFix({ config, props })) {
+    return config;
+  }
+
+  // check that swizzling is enabled; otherwise patch must be customized and applied manually
+  withInfoPlist(config, config => {
+    ensureFirebaseSwizzlingIsEnabled(config);
+    return config;
+  });
+
+  // apply patch
+  return withAppDelegate(config, config => {
+    return withOpenUrlFixForAppDelegate({ config, props });
+  });
+};
+
+// Interpret the plugin config to determine whether this fix should be applied
+export function shouldApplyIosOpenUrlFix({
+  config,
+  props,
+}: {
+  config: ExpoConfig;
+  props?: PluginConfigType;
+}): boolean {
+  const flag = props?.ios?.captchaOpenUrlFix;
+  if (flag === undefined || flag === 'default') {
+    // by default, apply the fix whenever 'expo-router' is detected in the same project
```

Comment:
I think this is a good choice, default Just Works, but people *can* disable it if they need to for whatever reason